### PR TITLE
Running Open Liberty on Amazon ECS

### PR DESCRIPTION
**Pull request by abutch3r**
https://blogs-draft-openlibertyio.mqj6zf7jocq.us-south.codeengine.appdomain.cloud/blog/2024/03/01/liberty-on-amazon-ecs.html

https://github.com/OpenLiberty/blogs/issues/2871

Replaces https://github.com/OpenLiberty/blogs/pull/3380

Expecting some updates - so will fix the date on the file as part of one of those

### CODE REVIEW

**Comment by gljrobinson** [comment]
Sorry for the delay on this @abutch3r, we've had a backlog of blogs to review and edit. I'm afraid I won't get round to reviewing this until the end of May/start of June, but I wanted to give you an update. Sorry for this.

File: posts/2024-07-30-liberty-on-amazon-ecs.adoc [hunk 1]
```diff
@@ -0,0 +1,265 @@
...
+
+Through this blog you've gained an understanding of the steps required to take a MicroProfile application running with Open Liberty and run and effectively scale it with Amazon ECS with AWS Fargate. Continue your learning by checking out some of the additional resources listed below.
+
+== Additional Resources
+* https://aws.amazon.com/ecs/[Amazon Elastic Container Service]
+* https://aws.amazon.com/fargate/[AWS Fargate]
+
+* https://docs.aws.amazon.com/AmazonECS/latest/developerguide/task_definitions.html[Amazon ECS Task Definitions]
+* https://docs.aws.amazon.com/AmazonECS/latest/bestpracticesguide/intro.html[Amazon ECS Best Practices]
+* https://docs.aws.amazon.com/elasticloadbalancing/latest/application/introduction.html[Amazon Elastic Load Balancing]
```
Comment (abutch3r) [inline posts/2024-07-30-liberty-on-amazon-ecs.adoc]: Given the cut down of the blog the follow links are now no longer relevant
- Amazon Elastic Load Balancing 
- Create an HTTPS listener for your Application Load Balancer
- AWS Permissions

While ALBs are referenced and created in the post - it is no longer to the depth that having the link is needed for as it is done within the definition of ECS

File: posts/2024-07-30-liberty-on-amazon-ecs.adoc [hunk 2]
```diff
@@ -0,0 +1,194 @@
...
+* <<Step 6: Create the Service>>
+* <<Step 7: Start the Service>>
+* <<Step 8: Access the Service>>
+* <<Step 9: Monitor the service>>
+* <<Step 10: Scale the application via auto-scaling policies>>
+
+== Deploying an Open Liberty Container to Amazon ECS
+
+=== Step 1: Ensure you have the correct pre-requisites
+To run a MicroProfile application on Amazon ECS we need to be able to create AWS resources and have an application to run. For this, you'll need to ensure you have an link:https://aws.amazon.com/[AWS] Administrator account. You'll also need link:https://www.docker.com/[Docker], link:https://git-scm.com/book/en/v2/Getting-Started-The-Command-Line[Git CLI], link:https://maven.apache.org/[Maven], and an externally accessible container registry that can host your images such as link:https://aws.amazon.com/ecr/[Amazon Elastic Container Registry (Amazon ECR)] or link:https://docs.docker.com/docker-hub/[Dockerhub].
```
Comment (dmuelle) [inline posts/2024-07-30-liberty-on-amazon-ecs.adoc]:
```suggestion
To run a MicroProfile application on Amazon ECS we need to be able to create AWS resources and have an application to run. Ensure you have the following prerequisites: 
* An link:https://aws.amazon.com/[AWS] Administrator account
* link:https://www.docker.com/[Docker]
* link:https://git-scm.com/book/en/v2/Getting-Started-The-Command-Line[Git CLI]
* link:https://maven.apache.org/[Maven]
* An externally accessible container registry that can host your images, such as link:https://aws.amazon.com/ecr/[Amazon Elastic Container Registry (Amazon ECR)] or link:https://docs.docker.com/docker-hub/[Dockerhub].
```

File: posts/2024-07-30-liberty-on-amazon-ecs.adoc [hunk 3]
```diff
@@ -0,0 +1,194 @@
...
+
+=== Step 2: Ensure you have a suitable containerized application
+If you don't already have a MicroProfile application of your own to use to try these steps out, then you can use the application provided in the 'finish' directory of the Open Liberty link:https://github.com/OpenLiberty/guide-getting-started/tree/prod/finish['Getting Started' guide]. A publicly accessible copy of the containerized application is available from: `icr.io/appcafe/open-liberty/samples/getting-started`.
```
Comment (dmuelle) [inline posts/2024-07-30-liberty-on-amazon-ecs.adoc]:
```suggestion
If you don't already have a MicroProfile application, you can use the application provided in the `finish` directory of the Open Liberty link:https://github.com/OpenLiberty/guide-getting-started/tree/prod/finish[Getting Started guide]. A publicly accessible copy of the containerized application is available from: `icr.io/appcafe/open-liberty/samples/getting-started`.
```

File: posts/2024-07-30-liberty-on-amazon-ecs.adoc [hunk 4]
```diff
@@ -0,0 +1,194 @@
...
+
+=== Step 3: Upload the container to a registry
+To use this 'getting started' image it needs to be uploaded to a internet accessible container registry. Amazon ECS supports a range of container registries outlined in their https://docs.aws.amazon.com/AmazonECS/latest/developerguide/task_definition_parameters.html#container_definition_image[Image Registry requirements]. As this blog requires an AWS account, you most likely already have access to link:https://aws.amazon.com/ecr/[Amazon ECR] to act as your registry. To upload the image use the https://docs.aws.amazon.com/AmazonECR/latest/userguide/getting-started-console.html[Get Started] instructions for Amazon ECR.
```
Comment (dmuelle) [inline posts/2024-07-30-liberty-on-amazon-ecs.adoc]:
```suggestion
Upload the `guide-getting-started` image to an internet accessible container registry. Amazon ECS supports a range of container registries outlined in their https://docs.aws.amazon.com/AmazonECS/latest/developerguide/task_definition_parameters.html#container_definition_image[Image Registry requirements]. As this blog requires an AWS account, you most likely already have access to link:https://aws.amazon.com/ecr/[Amazon ECR] to act as your registry. To upload the image, use the https://docs.aws.amazon.com/AmazonECR/latest/userguide/getting-started-console.html[Get Started] instructions for Amazon ECR.
```

File: posts/2024-07-30-liberty-on-amazon-ecs.adoc [hunk 5]
```diff
@@ -0,0 +1,194 @@
...
+
+=== Step 4: Create your Amazon ECS Cluster
+To create your Amazon ECS cluster follow step 1. in https://docs.aws.amazon.com/AmazonECS/latest/developerguide/getting-started-fargate.html[Getting started with the console using Linux containers on AWS Fargate].
```
Comment (dmuelle) [inline posts/2024-07-30-liberty-on-amazon-ecs.adoc]:
```suggestion
To create your Amazon ECS cluster, follow step 1. in https://docs.aws.amazon.com/AmazonECS/latest/developerguide/getting-started-fargate.html[Getting started with the console using Linux containers on AWS Fargate].
```

**Review by dmuelle — APPROVED** [review]
made some minor edits for structure and style, LGTM

**Pull request merged**